packages/loader/driver-utils/src/networkUtils.ts

```diff
@@ -6,18 +6,26 @@
 import { ITelemetryErrorEvent, ITelemetryLogger } from "@fluidframework/common-definitions";
 import { isOnline, OnlineStatus, canRetryOnError } from "./network";
 
+export function checkConnectionType() {
+    if (typeof navigator === "object" && navigator !== null) {
+        const nav = navigator as any;
+        const connection = nav.connection ?? nav.mozConnection ?? nav.webkitConnection;
+        if (connection !== null && typeof connection === "object") {
+            return String(connection.type);
+        }
+    }
+    return undefined;
+}
+
 export function logNetworkFailure(logger: ITelemetryLogger, event: ITelemetryErrorEvent, error?: any) {
```

Comment:
I'd suggest for logNetworkFailure to stop logging errors for all code paths (and simply use it here). Detection of offline is not reliable in browsers, so that path turned out to be dead end.

Reply:
Ok, interesting. I might do that in a quick follow-up PR to keep it separate from this, I'll take a look though.

Reply:
@vladsud - Actually did it now.  And Vivek and I found a bug in the logic anyway - event.online was sometimes a number, sometimes a string, but for the logging category calculation was compared v. the string so it was not correct sometimes.

The only other usage of logNetworkFailure is in parallel request code, and we agree with the change in semantics.